nextcord/timestamp.py

```diff
@@ -0,0 +1,70 @@
+from dataclasses import dataclass
+from datetime import datetime as dt
+
+
+@dataclass
+class Timestamp:
+    time: int
```

Comment:
maybe change the from_datetime to be a constructor?

```suggestion
class Timestamp:
    @overload
    def __init__(self, unix_time: int):
        ...
    
    def __init__(self, datetime: dt):
        ...
        
   def __init__(self, arg: Union[int, dt]):
       if isinstance(arg, int):
           self.time = arg
       elif isinstance(arg, dt):
           self.time = round(datetime.timestamp())
       else:
           raise TypeError("First argument must be an integer or a datetime object")
```

Reply:
This seems like a needlessly complex way to construct it when a classmethod works; it makes more sense to me as a user if to construct from a timestamp I need to call a classmethod, rather than assuming it will do magic in the constructor

Reply:
no need to assume, thats what the overloads are for. your IDE will tell u. generally when I want to construct an object the first place I check is the constructor